### PR TITLE
Remove bitcoin and legal forms

### PR DESCRIPTION
**Pull request by lgebhardt**
This cleans up the now unused PublisherLegalForms and Bitcoin related components. There's a lot of changes, almost all deletions. I've tried to group the changes by functionality into two commits, in the hopes that it makes it easier to review. There is a third commit, "Filter uphold_code and uphold_access_parameters from logs", which could be a standalone PR. I left it in here to avoid merge conflicts since we're also removing a bitcoin reference on the same line.

**Note**: Data is being removed from the database by dropping the PublisherLegalForms table and the encrypted bitcoin column from the Publishers table. If this data isn't stored elsewhere I can rework the migrations and/or code to preserve this, if needed.

### CODE REVIEW

File: app/controllers/api/publishers_controller.rb [hunk 1]
```diff
@@ -1,8 +1,4 @@
 class Api::PublishersController < Api::BaseController
-  before_action :require_verified_publisher,
-    only: %i(notify update_legal_form)
```
Comment (ayumi) [inline app/controllers/api/publishers_controller.rb]: should this filter still apply to `#notify`?

File: config/locales/en.yml [hunk 2]
```diff
@@ -205,7 +195,7 @@ en:
     save_and_continue: "Save and continue"
   static:
     landing_header: "Welcome to Brave Payments"
-    landing_body: "Brave users are already making contributions to thank you for your efforts. The first step toward collecting those contributions is to verify ownership of your web address so Brave users can see that your site is verified. Once fully verified, you will begin to receive daily micropayments in your Bitcoin wallet."
+    landing_body: "Brave users are already making contributions to thank you for your efforts. The first step toward collecting those contributions is to verify ownership of your web address so Brave users can see that your site is verified. Once fully verified, you will begin to receive daily micropayments in your Uphold account."
```
Comment (ayumi) [inline config/locales/en.yml]: Maybe we should prefer "Basic Attention Token wallet" – cc @mrose17

File: app/controllers/api/publishers_controller.rb (hunk 1 above)
Comment (lgebhardt) [reply app/controllers/api/publishers_controller.rb]: Sorry, this one slipped by.

Comment (lgebhardt) [reply app/controllers/api/publishers_controller.rb]: @ayumi ok, fixed

File: config/locales/en.yml (hunk 2 above)
Comment (ayumi) [reply config/locales/en.yml]: Tracking this at https://github.com/brave/publishers/issues/104

Comment (mrose17) [reply config/locales/en.yml]: @ayumi - agreed. i think we can use 'BAT wallet"

**Pull request merged**